src/mongocxx/database.cpp

```diff
@@ -157,5 +164,16 @@ collection database::collection(stdx::string_view name) const {
     return mongocxx::collection(*this, name);
 }
 
+const database::impl& database::get_impl() const {
+    if(!_impl) {
+        throw operation_exception{make_error_code(error_code::k_database_uninitialized)};
```

Comment:
Is operation_exception really the right type here? I thought that was meant to represent an operation that failed on the server. I think we either want to just throw mongocxx::exception here, or consider making a mongocxx::logic_error or similar that inherits from mongocxx::exception for this type of issue.